atlas-core/src/main/java/io/atlasmap/core/BaseAtlasModule.java

```diff
@@ -342,12 +348,14 @@ protected void processCombineField(AtlasSession session, Mapping mapping, List<F
                 addAudit(session, outputField.getDocId(), String
                         .format("Combine requires String field type for inputField.path=%s", inputField.getPath()),
                         outputField.getPath(), AuditStatus.WARN, null);
-                continue;
-            }
 
-            if (combineValues == null) {
-                // We need to support a sorted map w/ null values
-                combineValues = new HashMap<Integer, String>();
+                if (getConversionService().isPrimitive(inputField.getFieldType())) {
```

Comment:
ugh I forgot to mention this, can we look for a converter whenever intputField.getFieldType() != outputField.getFieldType()? did you hit any problem? (might be another bug then...)

Reply:
i.e. not just for primitives

Reply:
If converter doesn't exist exception is thrown and combine won't be performed. Would be ok to catch exception in BaseAtlasModule and log error when converter doesn't exist ?

Reply:
@mmelko +1, I think that's the best for now. Try the converter, if fails then catch the exception and fallback to object.toString(). It might be even better to put a WARN level audit if converter is not available.